### PR TITLE
Fix race condition in check-state-proof

### LINKED ISSUE #1052: Check-state-proof error
**Detailed Description**
Tested a scenario where the state proof verification should fail i.e. changed the transaction ID from what it was expected to be in the json response. The response does fail, however also throws an error statement.

```
Initializing state proof for transactionId: 0.0.88-1600124832-528679001 from source: /Users/simihunjan/Downloads/stateproof.json
Parsing address books...
4 node(s) found in address book
Parsed 1 address books
Parsing record file...
Parsed record, found 1 transactions
Parsing signature files...
Parsed 4 signature files
transactionId 0_0_88_1600124832_528679001 not present in recordFile. Transaction map is {0_0_88_1600124832_528679000}
StateProof validation for 0.0.88-1600124832-528679001 returned false
/Users/simihunjan/Downloads/hedera-mirror-node/hedera-mirror-rest/check-state-proof/utils.js:47
    if (err) throw err;
             ^

Error: ENOENT: no such file or directory, open '0_0_88_1600124832_528679001/apiResponse.json'

```

**Actual Behavior**
1. Use a transaction ID that is not in the record file and run the CLI command

**Expected Behavior**
The error lines should be removed 

**Environment:**
- Node: v11.8.0
- OS: Mac

cc: @atul-hedera

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Detailed description**:
- Fix HTTP not found warnings about UnhandledPromiseRejection
- Fix race condition in file/directory creation by using synchronous IO
- Change validation output to more clearly indicate state proof is valid
- Change logs to be more streamlined

**Which issue(s) this PR fixes**:
Fixes #1052 

**Special notes for your reviewer**:

* Unsuccessful output with http:
```
$ node index.js -t 0.0.88-1600124832-528679001 -h http://localhost:80

   ╭────────────────────────────────────────────────╮
   │                                                │
   │   Hedera Transaction State Proof Checker CLI   │
   │                                                │
   ╰────────────────────────────────────────────────╯

Initializing state proof for transaction ID 0.0.88-1600124832-528679001 from source: http://localhost:80/api/v1/transactions/0.0.88-1600124832-528679001/stateproof
Requesting stateproof files from http://localhost:80/api/v1/transactions/0.0.88-1600124832-528679001/stateproof...
Error fetching http://localhost:80/api/v1/transactions/0.0.88-1600124832-528679001/stateproof: Error: Not Found
```
* Successful output with http:
```
$ node --unhandled-rejections=strict index.js -t 0.0.88-1600124832-528679000 -h http://localhost:80 --unhandled-rejections=strict

   ╭────────────────────────────────────────────────╮
   │                                                │
   │   Hedera Transaction State Proof Checker CLI   │
   │                                                │
   ╰────────────────────────────────────────────────╯

Initializing state proof for transaction ID 0.0.88-1600124832-528679000 from file: http://localhost:80/api/v1/transactions/0.0.88-1600124832-528679000/stateproof
Requesting stateproof files from http://localhost:80/api/v1/transactions/0.0.88-1600124832-528679000/stateproof...
Supporting files and API response for the state proof will be stored in the directory 0_0_88_1600124832_528679000
4 node(s) found in address book
Parsed 1 address books
Parsed record, found 1 transactions
Parsed 4 signature files
Matching transaction was found in record file
Verify signatures passed for node 0.0.4
Verify signatures passed for node 0.0.3
Verify signatures passed for node 0.0.6
Verify signatures passed for node 0.0.5
Record file hash was successfully matched with signature files
-----------------------------------------------
The state proof is cryptographically valid
-----------------------------------------------
```
*Unsuccessful output with file:
```
$ node index.js -t 0.0.88-1600124832-528679001 -f 0_0_88_1600124832_528679000/apiResponse.json

   ╭────────────────────────────────────────────────╮
   │                                                │
   │   Hedera Transaction State Proof Checker CLI   │
   │                                                │
   ╰────────────────────────────────────────────────╯

Initializing state proof for transaction ID 0.0.88-1600124832-528679001 from file: 0_0_88_1600124832_528679000/apiResponse.json
Supporting files and API response for the state proof will be stored in the directory 0_0_88_1600124832_528679001
4 node(s) found in address book
Parsed 1 address books
Parsed record, found 1 transactions
Parsed 4 signature files
Transaction ID 0_0_88_1600124832_528679001 not present in record file. Available transaction IDs: 0_0_88_1600124832_528679000
-----------------------------------------------
The state proof is cryptographically invalid
-----------------------------------------------
```
* Successful output with file:
```
node index.js -t 0.0.88-1600124832-528679000 -f 0_0_88_1600124832_528679000/apiResponse.json

   ╭────────────────────────────────────────────────╮
   │                                                │
   │   Hedera Transaction State Proof Checker CLI   │
   │                                                │
   ╰────────────────────────────────────────────────╯

Initializing state proof for transaction ID 0.0.88-1600124832-528679000 from file: 0_0_88_1600124832_528679000/apiResponse.json
Supporting files and API response for the state proof will be stored in the directory 0_0_88_1600124832_528679000
4 node(s) found in address book
Parsed 1 address books
Parsed record, found 1 transactions
Parsed 4 signature files
Matching transaction was found in record file
Verify signatures passed for node 0.0.4
Verify signatures passed for node 0.0.3
Verify signatures passed for node 0.0.6
Verify signatures passed for node 0.0.5
Record file hash was successfully matched with signature files
-----------------------------------------------
The state proof is cryptographically valid
-----------------------------------------------
```

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

**Review by SimiHunjan — DISMISSED** [review]
Looks good!

**Review by xin-hedera — DISMISSED** [review]
LGTM

File: hedera-mirror-rest/check-state-proof/startUp.js [hunk 1]
```diff
@@ -97,11 +97,11 @@ const startUpScreen = () => {
       host = options.host;
     }
 
-    url = `${host}/api/v1/transactions/${transactionId}/stateproof`;
+    url = `http://${host}/api/v1/transactions/${transactionId}/stateproof`;
```
Comment (Nana-EC) [inline hedera-mirror-rest/check-state-proof/startUp.js:100]: when host is undefined and local/previewnet/testnet/mainnet hosts are used won't this result in `http://https....`

Comment (steven-sheehy) [reply hedera-mirror-rest/check-state-proof/startUp.js]: Thanks, reverted

**Review by Nana-EC — APPROVED** [review]
LGTM

**Pull request merged**